app/controllers/organisation_relationships_controller.rb

```diff
@@ -93,24 +93,37 @@ def create_managing_agent
     redirect_to managing_agents_organisation_path(related_organisation_id:)
   end
 
+  def remove_housing_provider
+    render "organisation_relationships/remove_housing_provider", layout: "application"
+  end
+
+  def delete_housing_provider
+    organisation_relationship_to_remove = OrganisationRelationship.find_by!(child_organisation_id: @organisation.id, parent_organisation_id: organisation_to_remove_id, relationship_type: OrganisationRelationship::OWNING)
+    delete(organisation_relationship_to_remove)
+    redirect_to housing_providers_organisation_path(removed_organisation_id: organisation_to_remove_id)
```

Comment:
We can use flash notices instead of passing the removed organisation in: https://api.rubyonrails.org/classes/ActionDispatch/Flash.html

Let me know if you'd like to chat about how this works.